test/shared_utils/compare.py

```diff
@@ -5,13 +5,43 @@
 This module focuses on comparing output specifically with xarray.
 """
 
+import json
 from itertools import count
 
-
 from xarray.backends.api import open_groups
 from xarray.core.datatree import DataTree
 from xarray import Dataset
 
+from reference_hashing import get_hashes_from_xarray_input
+
+
+SKIPPED_VARIABLES_OR_GROUPS = {
+    '/subset_files'  # SAMBAH tests - different UAT vs production
+}
+
+
```

Comment:
Some of the function names, like this one, are a bit clunky, referring to `xarray`. That's because we'll likely need to write other wrapper for things like HDF-EOS2 files that don't play nicely with `xarray`. (Possibly also GeoTIFFs)

Reply:
You could call out the type of file it works with netcdf4 or hdf5?
`nc4_matches_reference_hash_file` and `h5_matches_reference_hash_file`?

Reply:
I just looked in the notebook where this is used and it needs a better name.
`compare_nc4_to_hash_file` or similar.

Reply:
I was thinking of a thin wrapper, I feel like the using xarray isn't the important piece.  No to a giant if/else chain.